crates/ruff_python_parser/src/parser/statement.rs

```diff
@@ -100,69 +100,96 @@ impl<'src> Parser<'src> {
     fn parse_single_simple_statement(&mut self) -> Stmt {
         let stmt = self.parse_simple_statement();
 
         let has_eaten_semicolon = self.eat(TokenKind::Semi);
         let has_eaten_newline = self.eat(TokenKind::Newline);
 
         if !has_eaten_newline && !has_eaten_semicolon && self.at_simple_stmt() {
-            let range = self.current_token_range();
+            // test_err simple_stmts_on_same_line
+            // a b
+            // a + b c + d
+            // break; continue pass; continue break
             self.add_error(
-                ParseErrorType::SimpleStmtsInSameLine,
-                stmt.range().cover(range),
+                ParseErrorType::SimpleStatementsOnSameLine,
+                self.current_token_range(),
             );
         }
 
         if !has_eaten_newline && self.at_compound_stmt() {
+            // test_err simple_and_compound_stmt_on_same_line
+            // a; if b: pass; b
             self.add_error(
-                ParseErrorType::SimpleStmtAndCompoundStmtInSameLine,
-                stmt.range().cover(self.current_token_range()),
+                ParseErrorType::SimpleAndCompoundStatementOnSameLine,
+                self.current_token_range(),
             );
         }
 
         stmt
     }
 
+    /// Parses a sequence of simple statements.
+    ///
+    /// If there is more than one statement in this sequence, it is expected to be separated by a
+    /// semicolon. The sequence can optionally end with a semicolon, but regardless of whether
+    /// a semicolon is present or not, it is expected to end with a newline.
+    ///
+    /// Matches the `simple_stmts` rule in the [Python grammar].
+    ///
+    /// [Python grammar]: https://docs.python.org/3/reference/grammar.html
     fn parse_simple_statements(&mut self) -> Vec<Stmt> {
         let mut stmts = vec![];
-        let start = self.node_start();
         let mut progress = ParserProgress::default();
 
         loop {
             progress.assert_progressing(self);
+
             stmts.push(self.parse_simple_statement());
 
             if !self.eat(TokenKind::Semi) {
                 if self.at_simple_stmt() {
-                    for stmt in &stmts {
-                        self.add_error(ParseErrorType::SimpleStmtsInSameLine, stmt.range());
-                    }
+                    // test_err simple_stmts_on_same_line_in_block
+                    // if True: break; continue pass; continue break
+                    self.add_error(
+                        ParseErrorType::SimpleStatementsOnSameLine,
+                        self.current_token_range(),
+                    );
                 } else {
+                    // test_ok simple_stmts_in_block
+                    // if True: pass
+                    // if True: pass;
+                    // if True: pass; continue
+                    // if True: pass; continue;
+                    // x = 1
                     break;
                 }
```

Comment:
Nit: I think that's covered by the same check on line 197 (but I'm okay keeping it)